cdap-security/src/main/java/co/cask/cdap/security/authorization/DefaultAuthorizationEnforcer.java

```diff
@@ -43,12 +49,20 @@
 
   private final AuthorizerInstantiator authorizerInstantiator;
   private final boolean propagatePrivileges;
+  private final Principal masterUser;
 
   @Inject
   DefaultAuthorizationEnforcer(CConfiguration cConf, AuthorizerInstantiator authorizerInstantiator) {
     super(cConf);
     this.propagatePrivileges = cConf.getBoolean(Constants.Security.Authorization.PROPAGATE_PRIVILEGES);
     this.authorizerInstantiator = authorizerInstantiator;
+    String masterPrincipal = SecurityUtil.getMasterPrincipal(cConf);
+    try {
+      this.masterUser = masterPrincipal == null ? null : new Principal(new KerberosName(masterPrincipal).getShortName(),
+                                                                       Principal.PrincipalType.USER);
+    } catch (IOException e) {
```

Comment:
It doesn't seem like that constructor throws IOException.
(IOException is a checked exception, which the constructor doesn't declare as throws).